tests/unit/services/apollo-test.ts

```diff
@@ -18,23 +21,26 @@ module('Unit | Service | apollo', function(hooks) {
   });
 
   test('it uses clientOptions', function(assert) {
-    let customDataIdFromObject = o => o.name;
-    this.owner.register(
-      'service:overridden-apollo',
-      ApolloService.extend({
-        // Override the clientOptions.
-        clientOptions: computed(function() {
-          let opts = this._super(...arguments);
-          opts.dataIdFromObject = customDataIdFromObject;
-          return opts;
-        }),
-      })
-    );
-    let service = this.owner.lookup('service:overridden-apollo');
+    const dataIdFromObject = (o: { name: string }) => o.name;
+
+    class OverriddenApolloService extends ApolloService {
+      // Override the clientOptions.
+      @computed
+      get clientOptions() {
+        const { get } = getSuperComputed(this, 'clientOptions');
```

Comment:
This is interesting. I wonder if this needs to be a computed property at all, if it doesn't then we could use use `super.clientOptions`.

Reply:
I agree that these acrobatics are not really ideal. But I understand why it was implemented this way (see above comments). Maybe we can still find an alternative to computed.